images/win/scripts/Installers/Install-AzureCli.ps1

```diff
@@ -3,13 +3,32 @@
 ##  Desc:  Install Azure CLI
 ################################################################################
 
-Write-Host "Install the latest Azure CLI release"
-$azCliUrl = "https://aka.ms/installazurecliwindowsx64"
-Install-Binary -Url $azCliUrl -Name "azure-cli.msi"
+Write-Host 'Install the latest Azure CLI release'
+
+$azureCliConfigPath = 'C:\azureCli'
+# Store azure-cli cache outside of the provisioning user's profile
+[Environment]::SetEnvironmentVariable('AZURE_CONFIG_DIR', $azureCliConfigPath, [System.EnvironmentVariableTarget]::Machine)
+# make variable to be available in the current session
+${env:AZURE_CONFIG_DIR} = $azureCliConfigPath
+
+$azCliUrl = 'https://aka.ms/installazurecliwindowsx64'
+Install-Binary -Url $azCliUrl -Name 'azure-cli.msi'
 
 $azureCliExtensionPath = Join-Path $Env:CommonProgramFiles 'AzureCliExtensionDirectory'
-$null = New-Item -ItemType "Directory" -Path $azureCliExtensionPath
+$null = New-Item -ItemType 'Directory' -Path $azureCliExtensionPath
+
+[Environment]::SetEnvironmentVariable('AZURE_EXTENSION_DIR', $azureCliExtensionPath, [System.EnvironmentVariableTarget]::Machine)
+# make variable to be available in the current session
+${env:AZURE_EXTENSION_DIR} = $azureCliExtensionPath
+
+# Warm-up Azure CLI
+
+Write-Host "Warmup 'az'"
 
-[Environment]::SetEnvironmentVariable("AZURE_EXTENSION_DIR", $azureCliExtensionPath, [System.EnvironmentVariableTarget]::Machine)
+az --help
+if ($LASTEXITCODE -ne 0)
+{
+   throw "Command 'az --help' failed"
+}
```

Comment:
Don't we have tests that do the same?

Reply:
even if we do, it would be a leaked abstraction.
tests do testing, warmup does warmup

Reply:
Current test does `az --version`. Which is nice too, as it causes the caching of the version information on disk (as `az --version` does an update check and caches the result).

Reply:
someone once may want to change tests behavior. I would not like to depend on those changes, thus dedicated warm up makes sense (even if it's identical)

Reply:
Ok. I would only hide az's output and put braces according to1TBS rule.

Reply:
I used VS Code with whatever we have in .vscode, I wonder why do you think code is not compliant